### PR TITLE
feat: add wifi/eth bar component

### PR DESCRIPTION
**Pull request by notblam**

### CODE REVIEW

File: GlazeWM.Bar/Components/NetworkComponentViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          var primaryIntfGuid = list.InterfaceInfo[0].InterfaceGuid;
+
+          // Get wireless connection details.
+          var getType = CorrespondingTypeAttribute.GetCorrespondingTypes(WLAN_INTF_OPCODE.wlan_intf_opcode_current_connection, CorrespondingAction.Get).FirstOrDefault();
+          var interfaceDetails = WlanQueryInterface(hWlan, primaryIntfGuid, WLAN_INTF_OPCODE.wlan_intf_opcode_current_connection, default, out var dataSize, out var data, out var type);
+          if (interfaceDetails.Failed)
+            return _config.IconNoInternet;
+
+          // Get RSSI signal strength of connection.  
+          var connectionAttributes = (WLAN_CONNECTION_ATTRIBUTES)data.DangerousGetHandle().Convert(dataSize, getType);
```
Comment (lars-berger) [inline GlazeWM.Bar/Components/NetworkComponentViewModel.cs:70]: it's hard to follow what's going on here IMO

could we move the bulk of this out to `Infrastructure` eg a `NetworkService` with methods like `GetSignalStrength` etc?

File: GlazeWM.Domain/UserConfigs/NetworkComponentConfig.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    /// Text to display.
+    /// </summary>
+    public string IconWifiSignal0 { get; set; } = "";
+    public string IconWifiSignal25 { get; set; } = "";
+    public string IconWifiSignal50 { get; set; } = "";
+    public string IconWifiSignal75 { get; set; } = "";
+    public string IconWifiSignal100 { get; set; } = "";
+    public string IconEthernet { get; set; } = "";
+    public string IconNoInternet { get; set; } = "";
+    public string IconFontFamily { get; set; }
```
Comment (lars-berger) [inline GlazeWM.Domain/UserConfigs/NetworkComponentConfig.cs:18]: could we change this to:
```
label_disconnected

label_wifi_strength_0
label_wifi_strength_25
label_wifi_strength_50
label_wifi_strength_75
label_wifi_strength_100

label_eth_strength_0
label_eth_strength_25
label_eth_strength_50
label_eth_strength_75
label_eth_strength_100
```
when we add support for mixed fonts/colors, this means the user config could look something like: 
```yaml
label_wifi_strength_0: "{Material Icons:""} {ip_address} {wifi_name}"
label_wifi_strength_25: "{Material Icons:""} {ip_address} {wifi_name}"
label_wifi_strength_50:  "{Material Icons:""} {ip_address} {wifi_name}"
label_wifi_strength_75:  "{Material Icons:""} {ip_address} {wifi_name}"
label_wifi_strength_100:  "{Material Icons:""} {ip_address} {wifi_name}"

label_disconnected: "{Material Icons:""}"
```
for this PR, this means removing changing it so there's just a single `<TextBlock>` that conditionally shows the text for either `LabelWifiStrength0`, `LabelWifiStrength25` etc. Essentially how the current battery component works

File: GlazeWM.Bar/Components/NetworkComponentViewModel.cs (hunk 1 above)
Comment (notblam) [reply GlazeWM.Bar/Components/NetworkComponentViewModel.cs:70]: Agreed we can abstract some of this away.

File: GlazeWM.Infrastructure/WindowsApi/NetworkService.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      catch (PingException)
+      {
+        return false;
+      }
+      finally
+      {
+        pinger?.Dispose();
+      }
+      return pingable;
+    }
```
Comment (lars-berger) [inline GlazeWM.Infrastructure/WindowsApi/NetworkService.cs:54]: can be simplified

```csharp
try { 
     // Use Google DNS servers to check if online.
    var reply = new Ping().Send("8.8.8.8");
    return reply.Status == IPStatus.Success;
}
catch (Exception) {
    return false;
}

```

File: GlazeWM.Domain/UserConfigs/NetworkComponentConfig.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+namespace GlazeWM.Domain.UserConfigs
+{
+  public class NetworkComponentConfig : BarComponentConfig
+  {
+    /// <summary>
+    /// Text to display.
+    /// </summary>
+    /// <summary>
+    /// Text to display.
+    /// </summary>
```
Comment (lars-berger) [inline GlazeWM.Domain/UserConfigs/NetworkComponentConfig.cs:10]: repeated comment

File: GlazeWM.Infrastructure/WindowsApi/NetworkService.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+using System.Text;
+using Vanara.InteropServices;
+using static Vanara.PInvoke.IpHlpApi;
+using static Vanara.PInvoke.WlanApi;
+
+namespace GlazeWM.Infrastructure.WindowsApi
+{
+  public static class NetworkService
+  {
+    public static int getWlanRSSI()
```
Comment (lars-berger) [inline GlazeWM.Infrastructure/WindowsApi/NetworkService.cs:13]: method names should always be PascalCase (here and some cases in `NetworkComponentViewModel.cs`)

File: GlazeWM.Infrastructure/WindowsApi/NetworkService.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        return -1;
+      var primaryIntfGuid = list.InterfaceInfo[0].InterfaceGuid;
+
+      // Get wireless connection details.
+      var getType = CorrespondingTypeAttribute.GetCorrespondingTypes(WLAN_INTF_OPCODE.wlan_intf_opcode_current_connection, CorrespondingAction.Get).FirstOrDefault();
+      var interfaceDetails = WlanQueryInterface(hWlan, primaryIntfGuid, WLAN_INTF_OPCODE.wlan_intf_opcode_current_connection, default, out var dataSize, out var data, out _);
+      if (interfaceDetails.Failed)
+        return -1;
+
+      //   _currentSSID = connectionAttributes.strProfileName;
```
Comment (lars-berger) [inline GlazeWM.Infrastructure/WindowsApi/NetworkService.cs:27]: remove commented code

File: GlazeWM.Infrastructure/DependencyInjection.cs [hunk 7]
```diff
@@ -14,6 +14,7 @@ public static IServiceCollection AddInfrastructureServices(this IServiceCollecti
       services.AddSingleton<Bus>();
       services.AddSingleton<KeybindingService>();
       services.AddSingleton<WindowEventService>();
+      services.AddSingleton<NetworkService>();
```
Comment (lars-berger) [inline GlazeWM.Infrastructure/DependencyInjection.cs:17]: since `NetworkService` is static, it doesn't need to be registered in DI
